projects/control-service/cicd/.gitlab-ci.yml

```
@@ -249,7 +250,7 @@ control_service_vdk_heartbeat_release:
   stage: release
   script:
     - export CHART_NAME=pipelines-control-service
-    - export CHART_VERSION=$(cat projects/control-service/projects/helm_charts/$CHART_NAME/version.txt)
+    - export CHART_VERSION=$(cat projects/control-service/projects/helm_charts/$CHART_NAME/version.txt | grep -o '[0-9]\.[0-9]').$CI_PIPELINE_ID
```

That would not work if verison if 1.22 or 22.1 (double digits) or 1.22.2.3 . 

Maybe this will:

```suggestion
    - export CHART_VERSION=$(cat projects/control-service/projects/helm_charts/$CHART_NAME/version.txt | grep -o '^[0-9]\+\.[0-9]\+').$CI_PIPELINE_ID
```

---

Good point. I will actually omit the `^` at the start to allow whitespaces at the beginning of the version.

---

Actually, the last example still won't work without `^`.

---

I think it is more likely for people to leave a space than to use 4 versions. But either way is fine.